docs/src/developer_docs/model_interface.md

```diff
@@ -0,0 +1,412 @@
+  The default fallback returns `nothing`, which skips distributed synchronization.
+  Models with a `grid` property typically override this as
+  `architecture(model::MyModel) = model.grid.architecture`.
+
+- `timestepper(model)`: return the model's time-stepper object (or `nothing` if
+  the model does not use a time-stepper). Simulation uses this to reset the
+  time-stepper state when `reset!(sim)` is called. The default fallback returns
+  `nothing`, so models without a time-stepper need not implement this method.
+  Models with a time-stepper should implement this as
+  `timestepper(model::MyModel) = model.timestepper`.
+
+- `fields(model)` and `prognostic_fields(model)`: return `NamedTuple`s of
+  fields. These are not used by Simulation itself, but are conventions used
+  elsewhere in the ecosystem. `prognostic_fields(model)` should return the
+  fields that are time-stepped (used by NaN checkers and some diagnostics).
+  `fields(model)` should return a `NamedTuple` that includes both prognostic
+  fields and other fields that users might want to access or output, such as
+  pressure or diagnostic fields computed during `update_state!`.
+
+- `default_nan_checker(model)`: customize the `NaNChecker` that Simulation adds
+  by default.
+
+## Models that implement this interface
+
+Several models across the CliMA ecosystem implement this interface:
+
+**Oceananigans.jl**
+- `NonhydrostaticModel`: solves the incompressible Navier-Stokes equations
+- `HydrostaticFreeSurfaceModel`: solves the hydrostatic Boussinesq equations with a free surface
+- `ShallowWaterModel`: solves the shallow water equations
+
+**ClimaOcean.jl**
+- `OceanSeaIceModel`: couples ocean, sea ice, and atmosphere components for Earth system modeling.
+                      The components themselves may be `Simulation` that contain `AbstractModel`,
+                      generating a nested structure.
+
+**ClimaSeaIce.jl**
+- `SeaIceModel`: simulates sea ice thermodynamics and dynamics
+
+**Breeze.jl**
+- `AtmosphereModel`: simulates atmospheric dynamics
+
+## Example: a zero-dimensional `LorenzModel`
+
+The example below shows a deliberately tiny model that integrates the Lorenz
+system. The implementation demonstrates how little is required: store a `clock`,
+provide `time_step!` and `update_state!` implementations, and rely on fallbacks
+for the rest. Note that this model has no grid and no fields—just simple scalar
+state variables `x`, `y`, and `z`.
+
+### Implementing the interface
+
+```@example model_interface
+using Oceananigans
+using Oceananigans.Models: AbstractModel
+using Oceananigans.Simulations: Simulation, run!
+using Oceananigans.TimeSteppers: Clock, tick!
+using Oceananigans: TendencyCallsite, UpdateStateCallsite
+
+import Oceananigans.TimeSteppers: update_state!, time_step!
+
+mutable struct LorenzModel{FT} <: AbstractModel{Nothing, Nothing}
+    clock :: Clock{FT}
+    σ :: FT
```

Comment:
I actually liked it a bit better the way it was before :see_no_evil:, I was only teasing, I didn't mean to suggest that you had to change it :sweat_smile: 

but it's also fine to keep it like this, it's a simple model anyway.

Reply:
done